kratos/tests/auxiliar_files_for_python_unittest/materials_files/materials_custom_parameters.json

```diff
@@ -0,0 +1,64 @@
+{
+    "properties": [{
+        "model_part_name": "Main.Inlets",
+        "properties_id": 1,
+        "Material": {
+            "name": "steel",
+            "constitutive_law": {
+                "name": "LinearElastic3DLaw"
+            },
+            "Variables": {
+                "YOUNG_MODULUS": 200.0,
+                "POISSON_RATIO": 0.3,
+                "YIELD_STRESS": 400.0,
+                "CONFIGURATION_PARAMETERS" : {
```

Comment:
@thelfer this is the main difference, here I only put ""test" : true", but you can put literally anything and it will be saved in the CONFIGURATION_PARAMETERS variable

Reply:
If I may, I would suggest a more explicit name such as `MATERIAL_PROPERTIES` (or `CONSTITUTIVE_LAW_MATERIAL_PROPERTIES` or even `MFRONT_MATERIAL_PROPERTIES` thought this is not what I prefer).

I also would like to have a variable for the parameters of the behaviour (parameters are distinct from material properties in `MFront` in the sense that they  have default values and are stored in some global region, meaning that one can't have non constant and non uniform parameters). The name of this variable could be  `PARAMETERS` (or `CONSTITUTIVE_LAW_PARAMETERS` or even `MFRONT_PARAMETERS`).

Reply:
I would prefer  PARAMETERS, as is more generic, and this can be used for any pourpose

Reply:
Done

Reply:
So you added `PARAMETERS` as a new variable, great !
What about `MATERIAL_PROPERTIES` ?

Reply:
I prefer PARAMETERS, is more generi. It can be used in any context, not only in properties

Reply:
As you wish :).

I am currently implementing a dictionnary that could be used to map some TFEL glossary' names to Kratos' registred variable names. Let me give you some details about this.

Every variable in input and ouptut of a MFront constitutive law has a so-called external variables. Those are the names retrieved by `MGIS`. 

External names are the basis  of the "dependency management algorithm" described in the paper that I sent you.

TFEL glossary names are used by developpers of MFront' constitutive law to unambiguously tell to the constitutive law' user what is the expected variable. For example, a variable having the external name `Temperature` stands for the temperature (normally in Kelvin).... See http://tfel.sourceforge.net/glossary.html for the current list of entries in the `TFEL`' glossary.

The TFEL'  `Glossary` class introduces static variables such as `Glossary::Temperature` which are implicitly convertible to `std::string`. This allows avoiding typos.

The idea of external name is thus very close to the idea of registred variables in `Kratos`. The main difference is an `MFront` variable may have an external name which is not part of the glossary which adds a lot of flexibility.

But, this also means that I can create a mapping between the TFEL'glossary names and Kratos' variables names. My current implementation looks like:

```c++
      add("YoungModulus", YOUNG_MODULUS);
      add("YoungModulus1", YOUNG_MODULUS_X);
      add("YoungModulus2", YOUNG_MODULUS_Y);
      add("YoungModulus3", YOUNG_MODULUS_Z);
```

This way, if a MFront consistitutive law' material property has an external name which maps to a Kratos' variable, I would read it just like a native constitutive law. If not, I will use your newly introduced variable. 

This will mitigates the difference of approaches between MFront` and `Kratos`, don't you think ?

Reply:
For me this looks fine. I assume that the faster way is to make that dictionary static, so it will be defined on compilation time. Any further suggestion @RiccardoRossi?

Reply:
The mapping is currently in singleton built on first call.